packages/api-fetch/src/middlewares/preloading.js

```diff
@@ -45,8 +45,28 @@ function createPreloadingMiddleware( preloadedData ) {
 			const method = options.method || 'GET';
 			const path = getStablePath( options.path );
 
-			if ( parse && 'GET' === method && cache[ path ] ) {
-				return Promise.resolve( cache[ path ].body );
+			if ( 'GET' === method && cache[ path ] ) {
+				return Promise.resolve(
+					parse
+						? cache[ path ].body
+						: {
+								headers: {
```

Comment:
Why not actually use `new Headers`?

Reply:
Or perhaps even a new Response object, https://developer.mozilla.org/en-US/docs/Web/API/Response/Response.

Reply:
I thought it would be too complex, but it's actually cleaner. Updated!